sync/change_syncer.go

```diff
@@ -14,44 +14,46 @@ import (
 
 const channelSize = 10
 
-type applyChanges func(context.Context, <-chan change.Change) error
-type getChanges func(context.Context, chan<- change.Change) error
-
 // Syncer is responsible for sending incoming changes to the apply function
 type Syncer interface {
-	SyncChanges(ctx context.Context) error
+	Run(context.Context) error
+}
+
+// Handler interface for getting and applying changes
+type Handler interface {
+	Run(context.Context, chan change.Change) error
 }
 
 type syncer struct {
-	getChanges   getChanges
-	applyChanges applyChanges
+	getter  Handler
+	applier Handler
 }
 
 // New Syncer taking get and apply functions
 func New(
-	getChanges getChanges,
-	applyChanges applyChanges,
+	getter Handler,
+	applier Handler,
 ) Syncer {
 	return &syncer{
-		getChanges:   getChanges,
-		applyChanges: applyChanges,
+		getter:  getter,
+		applier: applier,
 	}
 }
 
-// SyncChanges until one function errors
-func (c *syncer) SyncChanges(ctx context.Context) error {
+// Run the sync until one function errors
+func (c *syncer) Run(ctx context.Context) error {
```

Comment:
maybe we should get rid of the object and give run the two handlers

Reply:
Could be better and would allow us doing something like
```
Run(ctx context.Context, applier Handler, getters ...Handler) 
```
to allow parallel reading of namespaces afterwards